design-proposals/instancetype-vm-rollout-support.md

```diff
@@ -0,0 +1,158 @@
+# Overview
+
+The [VM Rollout Strategy](https://kubevirt.io/user-guide/operations/vm_rollout_strategies/) feature introduced in [KubeVirt 1.2](https://github.com/kubevirt/kubevirt/releases/tag/v1.2.0) now allows for specific changes made to a running VM to propagate to the VMI without a restart. This design proposal covers extending this support to cover changes made to the referenced instance type or preference of a running VM.
+
+## Motivation
+
+The initial implementation of the VM Rollout Strategy feature does not support instance types or preferences with their use blocked by the VM validation webhook. Given the ability to now plug additional vCPUs (as sockets) and memory into a running VMI it would be beneficial to VM owners to extend this support to also plug additional resources introduced when switching to a new instance type and/or preference.
+
+## Goals
+
+- Allow for suitable changes introduced by a new instance type and/or preference to be propagated to the running VMI
+
+## Non Goals
+
+- Support for attributes of an instance type or preference not already supported by the `liveUpdate` VM Rollout Strategy feature. For example GPUs, Host Devices, NUMA, disk buses etc
+- Moving the `revisionName` of an `InstancetypeMatcher` or `PreferenceMatcher` to the `Status` field of the VM. The work to extract this field into the `Status` of a VM will continue through a previously documented [issue](https://github.com/kubevirt/kubevirt/issues/10145) and [PR](https://github.com/kubevirt/kubevirt/pull/10229).
```

Comment:
Can you please elaborate on the second non-goal bullet? How does it related to the live-update design?